File: storage/db-core/src/main/java/com/coffee/gu/store/StoreEntity.java
package com.coffee.gu.store;

import com.coffee.gu.enums.StoreStatus;
import com.coffee.gu.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;

import java.util.List;

@Table(name = "store")
@Entity
public class StoreEntity extends BaseEntity {
    private String name;
    private String branchCode;
    @Enumerated(EnumType.STRING)
    private StoreStatus status;
    private String address;
    private Double latitude;
    private Double longitude;
    private String phoneNumber;
    private String representative;
    private String tradeName;
    private String businessRegistrationNumber;
    private String businessAddress;

    public StoreEntity() {}

    private StoreEntity(String name, String branchCode, StoreStatus status, String address, Double latitude, Double longitude, String phoneNumber, String representative, String tradeName, String businessRegistrationNumber, String businessAddress) {
        this.name = name;
        this.branchCode = branchCode;
        this.status = status;
        this.address = address;
        this.latitude = latitude;
        this.longitude = longitude;
        this.phoneNumber = phoneNumber;
        this.representative = representative;
        this.tradeName = tradeName;
        this.businessRegistrationNumber = businessRegistrationNumber;
        this.businessAddress = businessAddress;
    }

    public static StoreEntity from(Store store) {
        return new StoreEntity(
                store.getName(),
                store.getBranchCode(),
                store.getStatus(),
                store.getSalesInformation().location().address(),
                store.getSalesInformation().location().latitude(),
                store.getSalesInformation().location().longitude(),
                store.getSalesInformation().phoneNumber(),
                store.getBusinessInformation().representative(),
                store.getBusinessInformation().tradeName(),
                store.getBusinessInformation().businessRegistrationNumber(),
                store.getBusinessInformation().businessAddress()
        );
    }

    public Store toModel() {
        return new Store(
          this.id,
          this.name,
          this.branchCode,
          this.status,
          new SalesInformation(
            new StoreLocation(this.address, this.latitude, this.longitude),
            List.of(),
            this.phoneNumber
          ),
          new BusinessInformation(
            this.representative,
            this.tradeName,
            this.businessRegistrationNumber,
            this.businessAddress
          )
        );
    }

    public String getName() {
        return name;
    }

    public String getBranchCode() {
        return branchCode;
    }

    public StoreStatus getStatus() {
        return status;
    }

    public String getAddress() {
        return address;
    }

    public Double getLatitude() {
        return latitude;
    }

    public Double getLongitude() {
        return longitude;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public String getRepresentative() {
        return representative;
    }

    public String getTradeName() {
        return tradeName;
    }

    public String getBusinessRegistrationNumber() {
        return businessRegistrationNumber;
    }

    public String getBusinessAddress() {
        return businessAddress;
    }
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/store/StoreEntity.kt
package com.coffee.gu.store

import com.coffee.gu.BaseEntity
import com.coffee.gu.enums.StoreStatus
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Table(name = "store")
@Entity
class StoreEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,
    val name: String,
    val branchCode: String,
    @Enumerated(EnumType.STRING)
    val status: StoreStatus,
    val address: String,
    val latitude: Double,
    val longitude: Double,
    val phoneNumber: String,
    val representative: String,
    val tradeName: String,
    val businessRegistrationNumber: String,
    val businessAddress: String,
) : BaseEntity() {

    fun toModel(): Store {
        return Store(
            id = id,
            name = name,
            branchCode = branchCode,
            status = status,
            salesInformation = SalesInformation(
                location = StoreLocation(address, latitude, longitude),
                hours = emptyList(),
                phoneNumber = phoneNumber,
            ),
            businessInformation = BusinessInformation(
                representative = representative,
                tradeName = tradeName,
                businessRegistrationNumber = businessRegistrationNumber,
                businessAddress = businessAddress,
            ),
        )
    }

    companion object {
        fun from(store: Store): StoreEntity {
            val salesInfo = store.salesInformation ?: throw IllegalArgumentException("SalesInformation must not be null")
            val businessInfo = store.businessInformation ?: throw IllegalArgumentException("BusinessInformation must not be null")
            return StoreEntity(
                id = store.id,
                name = store.name,
                branchCode = store.branchCode,
                status = store.status,
                address = salesInfo.location.address,
                latitude = salesInfo.location.latitude,
                longitude = salesInfo.location.longitude,
                phoneNumber = salesInfo.phoneNumber,
                representative = businessInfo.representative,
                tradeName = businessInfo.tradeName,
                businessRegistrationNumber = businessInfo.businessRegistrationNumber,
                businessAddress = businessInfo.businessAddress,
            )
        }
    }
}
